chalice/pipeline.py

```diff
@@ -45,12 +38,14 @@ class CreatePipelineTemplate(object):
         "Outputs": {},
     }
 
-    def __init__(self):
-        # type: () -> None
-        pass
+    def __init__(self, codebuild_image=None):
+        # type: (Optional[str]) -> None
+        self._codebuild_image = codebuild_image
 
-    def _codebuild_image(self, lambda_python_version):
+    def _get_codebuild_image(self, lambda_python_version):
```

Comment:
I would expect the  `lambda_python_version` here be optional now since you could have provided an explicit codebuild image. I think this applies to the `create_template` method as well.